bundles/paikkatietoikkuna/coordinateconversion/components/table.js

```diff
@@ -0,0 +1,191 @@
+                var rowcounter = this.template.rowcounter({ rows: this.loc.utils.rows })
+                                                                                
+                var coordinatefield = this.template.table({  input: this.loc.coordinatefield.input,
+                                                                north:this.loc.coordinatefield.north,
+                                                                east:this.loc.coordinatefield.east,
+                                                                ellipse_height: "" });
+
+                var table = this.template.tableWrapper.clone();
+
+                table.append(coordinatefield);                
+
+                table.find('.coordinatefield-table h5').append(rowcounter);
+
+                this.setElement( table );
+
+
+                var coords = {};
+                for( var i = 0; i < 10; i++ ) {
+                    table.find("#oskari-coordinate-table").append(this.template.row( { coords: coords } ) );
+                    me.incrementNumRows();
+                }
+
+                return this.getContainer();
+        },
+        incrementNumRows: function () {
+                this.getContainer().find("#num").text( this.numrows++ );
+        },
+        decrementNumRows: function () {
+                this.getContainer().find("#num").text( this.numrows-- );
+        },
+        isEditable: function ( editable ) {
+            var rows = this.getElements().rows;
+            if( !editable ) {
+                rows.each( function ( row ) {
+                    jQuery(this).find('td').attr("contenteditable", false);
+                });
+            }
+            else {
+                rows.each( function ( row ) {
+                    jQuery(this).find('td').attr("contenteditable", true);
+                });
+            }
+        },
+        removeRow: function () {
+            this.getElements().rows.first().remove();
+            this.decrementNumRows();
+        },
+        /**
+         * @method populate
+         *
+         * {Object} data, each key need to have property lon & lat 
+         */
+        populate: function( cell, data ) {
+            var table = this.getElements().table;
+            var i = 0;
+            for ( var key in data ) {
+                if ( i <= 9 ) {
+                    if ( data.hasOwnProperty( key ) ) {
+                        var row = this.template.row( { coords: data[key] } );
+                        jQuery( cell ).parent().after( row );
+                        this.incrementNumRows();
+                        i++;
+                    }
+                }
+            }
+            table.trigger('rowCountChanged');
+        },
+        addRows: function (coords) {
+            var table = this.getElements().table;
+            for (var i = 0; i < coords.length; i++ ) {
+                var row = this.template.row( { coords: coords[i] } );
+                table.find('tr:first').after(row);
```

Comment:
Why tr:first? New row always comes to the top of the table? Maybe the function name should be renamed to indicate that.